src/lambda/commands/createNewSamApp.ts

```diff
@@ -56,43 +55,47 @@ export async function resumeCreateNewSamApp(activationLaunchPath: ActivationLaun
 }
 
 export interface CreateNewSamApplicationResults {
-    reason: 'unknown' | 'userCancelled' | 'fileNotFound' | 'complete' | 'error'
-    result: 'pass' | 'fail' | 'cancel'
     runtime: string
+    result: Result
 }
 
+type createReason = 'unknown' | 'userCancelled' | 'fileNotFound' | 'complete' | 'error'
+
 /**
  * Runs `sam init` in the given context and returns useful metadata about its invocation
  */
 export async function createNewSamApplication(
     channelLogger: ChannelLogger,
     samCliContext: SamCliContext = getSamCliContext(),
     activationLaunchPath: ActivationLaunchPath = new ActivationLaunchPath()
-): Promise<CreateNewSamApplicationResults> {
-    const results: CreateNewSamApplicationResults = {
-        reason: 'unknown',
-        result: 'fail',
-        runtime: 'unknown'
-    }
+): Promise<void> {
+    let createResult: Result = 'Succeeded'
+    let reason: createReason = 'unknown'
+    let createRuntime: Runtime | undefined
+    let config: CreateNewSamAppWizardResponse | undefined
+
+    let initArguments: SamCliInitArgs
```

Comment:
initArguments doesn't appear to have broadened use in this change. Why did it change from a const (line 95) to a let (here)?

Reply:
it needs to be used outside of the try block it is set in